cocos/render-scene/core/pass.ts

```diff
@@ -595,20 +591,25 @@ export class Pass {
         this._stage = RenderPassStage.DEFAULT;
         if (cclegacy.rendering && cclegacy.rendering.enableEffectImport) {
             const r = cclegacy.rendering;
-            this._phase = r.completePhaseName(info.phase);
-            this._passID = r.getPassID(info.pass);
+            if (typeof info.phase === 'number') {
+                this._passID = (info as Pass)._passID;
+                this._phaseID = (info as Pass)._phaseID;
```

Comment:
if info.phase is number, the info must be type of `Pass`. This is determined by the code.

Reply:
These codes only exists in ts?

Reply:
They will be added to native pipeline later.

Reply:
I haven't seen c++ code yet. I will properly handle these cast. Avoid them at best.